src/librustc_parse/parser/diagnostics.rs

```diff
@@ -282,6 +286,30 @@ impl<'a> Parser<'a> {
         Err(err)
     }
 
+    fn check_too_many_raw_str_terminators(&mut self, err: &mut DiagnosticBuilder<'_>) -> bool {
+        let prev_token_raw_str = match self.prev_token.kind {
+            TokenKind::Literal(Lit {
+                kind: LitKind::StrRaw(n) | LitKind::ByteStrRaw(n), ..
+            }) => Some(n),
+            _ => None,
+        };
+
+        if let Some(n_hashes) = prev_token_raw_str {
```

Comment:
```suggestion
        if let TokenKind::Literal(Lit { kind: LitKind::StrRaw(n_hashes) | LitKind::ByteStrRaw(n_hashes), .. }) = prev_token_raw_str {
```
Looks like the previous `match` can be merged into this `if let`.

Reply:
ended up just merging it into one match